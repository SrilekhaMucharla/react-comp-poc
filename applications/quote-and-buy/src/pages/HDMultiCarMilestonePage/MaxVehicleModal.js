import React from 'react';
import PropTypes from 'prop-types';
import { AnalyticsHDModal as HDModal } from '../../web-analytics';
import * as messages from './HDMultiCarMilestonePage.messages';
import { pageMetadataPropTypes } from '../../constant/propTypes';

const MaxVehicleModal = ({
    onCancel, onConfirm, onClose, show, pageMetadata
}) => (
    <HDModal
        webAnalyticsView={{ ...pageMetadata, page_section: messages.reachedMax }}
        webAnalyticsEvent={{ event_action: messages.reachedMax }}
        id="maximum-vehicle-count-modal"
        show={show}
        headerText={messages.important}
        cancelLabel={messages.noGoBack}
        confirmLabel={messages.ok}
        onCancel={onCancel}
        onConfirm={onConfirm}
        onClose={onClose}
    >
        <span>{messages.reachedMax}</span>
    </HDModal>
);

MaxVehicleModal.propTypes = {
    onCancel: PropTypes.func.isRequired,
    onConfirm: PropTypes.func.isRequired,
    onClose: PropTypes.func.isRequired,
    show: PropTypes.bool,
    pageMetadata: PropTypes.shape(pageMetadataPropTypes).isRequired
};

MaxVehicleModal.defaultProps = {
    show: false
};

export default MaxVehicleModal;
