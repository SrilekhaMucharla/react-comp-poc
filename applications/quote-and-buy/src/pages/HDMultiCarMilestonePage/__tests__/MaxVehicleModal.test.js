import React from 'react';
import { shallow } from 'enzyme';
import MaxVehicleModal from '../MaxVehicleModal';

describe('MaxVehicleModal', () => {
    // given
    const onCancel = jest.fn();
    const onConfirm = jest.fn();
    const onClose = jest.fn();
    const headerText = 'You\'ve reached the maximum amount of cars.';
    const id = 'maximum-vehicle-count-modal';

    const getWrapper = (props = {}) => shallow(
        <MaxVehicleModal
            onCancel={onCancel}
            onConfirm={onConfirm}
            onClose={onClose}
            headerText={headerText}
            show
            {...props} />
    );

    it('should render correctly and match the snapshot', () => {
        // given
        const wrapper = getWrapper();
        // then
        expect(wrapper).toMatchSnapshot();
    });

    it('should display HDModal', () => {
        // given
        const wrapper = getWrapper();
        // then
        expect(wrapper.find(`#${id}`)).toHaveLength(1);
    });

    it('should contain driver name', () => {
        // given
        const wrapper = getWrapper();
        // then
        expect(wrapper.find('span').text()).toContain(headerText);
    });

    it('should contain "this driver"', () => {
        // given
        const noDriverString = 'You\'ve reached the maximum amount of cars.';
        const wrapper = shallow(
            <MaxVehicleModal
                onCancel={onCancel}
                onConfirm={onConfirm}
                onClose={onClose}
                headerText=""
                show />
        );
        // then
        expect(wrapper.find('span').text()).toContain(noDriverString);
    });

    it('should pass functions to HDModal', () => {
        // given
        const wrapper = getWrapper();
        const modalProps = wrapper.find(`#${id}`).props();
        // then
        expect(modalProps.onCancel).toBe(onCancel);
        expect(modalProps.onClose).toBe(onClose);
        expect(modalProps.onConfirm).toBe(onConfirm);
    });
});
