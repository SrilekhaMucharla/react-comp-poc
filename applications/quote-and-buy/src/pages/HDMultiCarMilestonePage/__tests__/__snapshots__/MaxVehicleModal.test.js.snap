// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaxVehicleModal should render correctly and match the snapshot 1`] = `
<ForwardRef
  cancelLabel="Go back"
  confirmLabel="OK"
  headerText="Important"
  id="maximum-vehicle-count-modal"
  onBeforeOpen={null}
  onCancel={[MockFunction]}
  onClose={[MockFunction]}
  onConfirm={[MockFunction]}
  show={true}
  webAnalyticsEvent={
    Object {
      "event_action": "You've reached the maximum amount of cars.",
    }
  }
  webAnalyticsView={
    Object {
      "page_section": "You've reached the maximum amount of cars.",
    }
  }
>
  <span>
    You've reached the maximum amount of cars.
  </span>
</ForwardRef>
`;
